backend/middlewares/isAuthenticated.js

```diff
@@ -1,13 +1,17 @@
 import { HttpError } from "../helpers/HttpError.js";
 import { prisma } from "../lib/db.js";
+import jwt from "jsonwebtoken";
+
+const { JWT_SECRET } = process.env;
 
 export const isAuthenticated = async (req, res, next) => {
   try {
-    const { user } = req.cookies;
-    if (!user) {
+    const token = req.cookies.token;
+    if (!token) {
       return next(HttpError(401));
     }
-    const { id } = JSON.parse(user);
+    const { id } = jwt.verify(token, JWT_SECRET);
+
     const dbUser = prisma.user.findFirst({
```

Comment:
We don't need to check whether the user exists, for now. Since we don't have any functionality related to deleting users, we can be sure that user exists if the token is successfully validated.